### PR TITLE
feat(vite): update to vitest v1

### LINKED ISSUE #20580: outdated peer dependency in @nx/vite package.json
### Current Behavior

Currently there is an outdated peer dependency of vitest in nx/vite which hinders us from using latest stable version of vitest with nx.

### Expected Behavior

We can use the latest stable version of vitest.

### GitHub Repo

_No response_

### Steps to Reproduce

1. try to install vitest@latest or vitest@1.0.1 

### Nx Report

```shell
-
```

### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

"peerDependencies": {
    "vite": "^5.0.0",
    "vitest": ">=0.34.6 <1.0.0"
  },
  
  this is outdated

### LINKED ISSUE #20722: Support Vitest v1
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

The nx/vite package has a peer dependency on [Vitest < 1](https://github.com/nrwl/nx/blob/332b0120df97fc05e52e4156b01df0208fbb925e/packages/vite/package.json#L41) . It would be great to use the stable release instead.

## Motivation

Vitest v1 does come with a range of improvements. https://github.com/vitest-dev/vitest/releases/tag/v1.0.0

### PR DESCRIPTION
**Pull request by mandarini**
* Updating vitest to v1
* Added migration for coverage thresholds to support new syntax:
ref: https://github.com/vitest-dev/vitest/blob/9c552b6f8decb78677b20e870eb430184e0b78ea/packages/vitest/src/types/coverage.ts#L181
https://github.com/vitest-dev/vitest/blob/9c552b6f8decb78677b20e870eb430184e0b78ea/packages/vitest/src/types/coverage.ts#L225
```
coverage: {
      thresholds: {
        lines: 100,
        statements: 100,
        functions: 100,
        branches: 1000,
      },
    },
```

### CODE REVIEW

File: packages/vite/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
   },
   "peerDependencies": {
     "vite": "^5.0.0",
-    "vitest": ">=0.34.6 <1.0.0"
+    "vitest": "<=1.0.0"
```
Comment (JamesHenry) [inline packages/vite/package.json:41]:
```suggestion
    "vitest": ">=1.0.0"
```

Surely?

**Review by JamesHenry — CHANGES_REQUESTED** [review]
The changelog seemed pretty dense when I saw it a few days ago, were there no generated config changes (and associated migrations) necessary?

File: packages/vite/package.json (hunk 1 above)
Comment (mandarini) [reply packages/vite/package.json:41]: oh lol yes

**Comment by mandarini** [comment]
@JamesHenry works on my machine :P 
I mean, I tried it on a couple of repos locally and the tests seemed to run ok. For the next cycle, I will look more into any new features they have for monorepos or anything we do want to support. But just for running the tests with our executors, and the bare-minimum we generate, it seems to be working ok?

**Comment by JamesHenry** [comment]
Ok cool, we can see if CI can pass ok to be sure

**Review by jaysoo — APPROVED** [review]

**Review by Coly010 — APPROVED** [review]

**Review by xiongemi — APPROVED** [review]

**Review by JamesHenry — APPROVED** [review]

**Pull request merged**